### ISSUE TITLE
addSingletonFactory is broken

### ISSUE THREAD

**Issue by nsk-mironov** [kohesive/injekt#7]
```kotlin
public class Singleton(val count: Int) {
  init {
    println("not really a singleton $count")
  }
}

public class Magic() {
  public val first: Singleton by Delegates.injectValue()
  public val second: Singleton by Delegates.injectValue()
  public val third: Singleton by Delegates.injectValue()
}

public object Hello : InjektMain() {
  override fun InjektRegistrar.registerInjectables() {
    val counter = AtomicInteger()

    addSingletonFactory {
      Singleton(counter.incrementAndGet())
    }
  }

  platformStatic public fun main(args: Array<String>) {
    val test = Magic()

    println("first ${test.first.count}")
    println("second ${test.second.count}")
    println("third ${test.third.count}")
  }
}
```

Output:
```
not really a singleton 1
not really a singleton 2
not really a singleton 3
first 1
second 1
third 1
```

**Comment by apatrida** [comment]
Thanks, problem with the concurrentGetOrPut taken from Kara is broken, quick fix coming.

**Comment by apatrida** [comment]
Fixed in v `1.1.1` - releasing to maven central.

I will also release `1.1.1-JDK8` which uses a safe concurrentGetOrPut.  And the registry based on Guava coming soon is also safe.
